### ISSUE TITLE
Reproduce the statistical results in Table 2

### ISSUE THREAD

**Issue by syxd** [bdy9527/FAGCN#1]
Great work! 
We are reproducing the statistical results in Table 2 and want to know how to design the 10 runs mentioned in the article for FAGCN. Use random seeds [0,1,2,...,9] and fixed hyperparameters in Sec 5.2?

We have re-run the code with random seed=0,1,2,3 on the Cora dataset and use the same hyperparameters mentioned in Sec 5.2, but the test accuracy only get 0.803(0), 0.806(1), 0.809(2), 0.798(3) respectively. 

The torch we used is 1.5.1 which is same as mentioned in the "requirement.txt".
Do I miss something? Could you please help me to reproduce the results?

Thanks!

Related code:
```
# Hyperparameters setting
parser = argparse.ArgumentParser()
parser.add_argument('--dataset', default='cora')    # new_squirrel, syn0
parser.add_argument('--lr', type=float, default=0.01, help='Initial learning rate.')
parser.add_argument('--weight_decay', type=float, default=1e-3, help='Weight decay (L2 loss on parameters).')
parser.add_argument('--epochs', type=int, default=500, help='Number of epochs to train.')
parser.add_argument('--hidden', type=int, default=32, help='Number of hidden units.')
parser.add_argument('--dropout', type=float, default=0.6, help='Dropout rate (1 - keep probability).')
parser.add_argument('--eps', type=float, default=0.2, help='Fixed scalar or learnable weight.')
parser.add_argument('--layer_num', type=int, default=4, help='Number of layers')
parser.add_argument('--train_ratio', type=float, default=0.6, help='Ratio of training set')
parser.add_argument('--patience', type=int, default=100, help='Patience')
args = parser.parse_args()
```

```
# Reset random seed
seed = 0  # 0,1,2,3
np.random.seed(seed)
torch.manual_seed(seed)
torch.cuda.manual_seed(seed)
```

```
# Load dataset
data = dgl.data.CoraDataset()
edge = data.graph.edges
feat = data.features
labels = data.labels
index = np.arange(feat.shape[0])
train = index[data.train_mask.astype(bool)]
val = index[data.val_mask.astype(bool)]
test = index[data.test_mask.astype(bool)]
...
```

**Comment by bdy9527** [comment]
Sorry for the late reply.

I check your code and the source code in **FAGCN.zip**, and find that the difference is in the dataloader. Here are my code:

```
edge = np.loadtxt('../low_freq/{}.edge'.format(dataset), dtype=int).tolist()
feat = np.loadtxt('../low_freq/{}.feature'.format(dataset))
labels = np.loadtxt('../low_freq/{}.label'.format(dataset), dtype=int)
train = np.loadtxt('../low_freq/{}.train'.format(dataset), dtype=int)
val = np.loadtxt('../low_freq/{}.val'.format(dataset), dtype=int)
test = np.loadtxt('../low_freq/{}.test'.format(dataset), dtype=int)
...
```

May this can help you.

**Comment by syxd** [comment]
Thanks for your reply!

I find the "Cora" dataset in FAGCN.zip, and re-run the code with this cora data. The test accuracy gets 0.833 with random seed 0, which is much better than 0.803 in my previous experiment. 

The only difference between this experiment (FAGCN.zip) and my previous experiments (DGL) is the used cora data. Then I find the cora data provided by DGL and FAGCN.zip looks like different. Here are two pictures to compare the detailed information of two cora data files.
Cora data provide by DGL:
<img width="1347" alt="未命名1614238604" src="https://user-images.githubusercontent.com/22913490/109119624-e74bf980-777f-11eb-92e8-fbb6358a98eb.png">
Cora data provide by FAGCN.zip:
<img width="1350" alt="未命名1614238612" src="https://user-images.githubusercontent.com/22913490/109119597-da2f0a80-777f-11eb-91fc-1be118c9b5ec.png">

As we can see, the id of edges, the id of nodes and the size of validation data are different in two files.
I'm wondering why the test accuracy are different when using different cora data files and why the cora data in FAGCN.zip is different with the cora data in DGL? Could you please help me to find the reasons?

Thanks!

**Comment by bdy9527** [comment]
I guess your DGL version is not the latest, because you load Cora dataset through `dgl.data.CoraDataset()`.

I remember that when the DGL version < 0.5, DGL team use the Cora dataset in PyGCN (Pytorch version) rather than the original GCN (Tensorflow version). Unfortunately, the split of PyGCN is not the same as original GCN.

I found this bug and report it to the DGL team and they fix it in the latest version.

I suggest you update DGL into the latest version. Besides, I also test the Cora dataset in DGL through [this](https://docs.dgl.ai/api/python/dgl.data.html#citation-network-dataset):
```
data = CoraGraphDataset()
graph = data[0]
print(graph)
train_mask = graph.ndata['train_mask']
val_mask = graph.ndata['val_mask']
test_mask = graph.ndata['test_mask']
labels = graph.ndata['label']
feat = graph.ndata['feat']
nclass = len(set(labels.tolist()))

return graph, nclass, feat, labels, train_mask, val_mask, test_mask
```

With random seed 0 and DGL == 0.5.3, the test accuracy is 0.842.

May this help you.

**Comment by syxd** [comment]
I think this is the reason. Thanks a lot! 

Besides, Could you please tell me the random seeds you used or how do you apply 10 runs?
